Source/towners.cpp

```diff
@@ -742,6 +742,9 @@ void TalkToGirl(PlayerStruct &player, TownerStruct &girl)
 	case QUEST_ACTIVE:
 		InitQTextMsg(TEXT_GIRL3);
 		return;
+	case QUEST_DONE:
+		// do nothing
+		return;

```

Comment:
Didn't we mention about not allowing the player to select the NPC anymore? That felt like a more robust option than just doing nothing.

I'd rather see that implemented if possible.

Reply:
puh - are you sure? I guess this would be confusing.

Reply:
> I guess this would be confusing.

That's exactly why. People expect _something_ to happen when clicking on a highlighted element in the game. Literally every other entity interaction results in some effect, sound, or a combination of both. "Nothing happening" goes against that standard behavior and is really confusing.

On the other hand, objects which have already been interacted with and cannot produce any more results, stop being highlightable. I think extending it to NPCs would make sense here.

Again, I'm not particularly opposed to actually adding some dialog line for her, but disabling the interaction is a conceptually simpler change that avoids the confusion without having to introduce extra content to the game.

Reply:
i can do this but the changes would be more complex. should also the mouse over info be disabled?

Reply:
I think fixing the dialog logic would be a good, and important, first step. Improvements can come later. That way we don't do multiple things in one commit either.

Reply:
She mentions of going home, maybe completely remove her if she isn't supposed to interact.

Reply:
I can do this, if you wish but at first at least the farmer quest fix should be merged because it also would need the same deactivation.

Reply:
letting her go home, is also a nice idea.

Reply:
okay i added this.
But not seeing her makes the tree looking somewhat empty ;) and feels like if the game was just startet new

Reply:
additionally to this the girls sprite with the teddy will not be shown anymore (as it only is loaded on init if quest is done)

so maybe I tend to roll this change back

Reply:
so I reset the change. Imho it would be best to first just recreate the correct vanilla behaviour of not crying on Teddys return

Reply:
I think this speaks more to a deficiency in the quest log, which doesn't currently have a "completed quests" section. I have taken long breaks and come back to old saves with no memory of what quests I had completed. I think I spent a long time looking for black mushroom once when I had in fact already finished it.

Reply:
> I think this speaks more to a deficiency in the quest log, which doesn't currently have a "completed quests" section.

This feels like an obviously-good idea to me. Have you considered documenting the proposal as an issue?